### PR TITLE
[#6] Updated setup.py to create an entry points line

### PR DESCRIPTION
**Pull request by monicaruttle**
This removes the requirement to have a script in the bin folder. With this change, you can call the standup_notes package directly as a command line script after installing it using "python setup.py install". Once these changes are approved, I'll actually release to it PyPi and add instructions in the README on how to install it from there.

### CODE REVIEW

**Comment by monicaruttle** [comment]
If someone could test this, that'd be great. It worked for me however I had to mess around with where setup.py installs to, but I think that might just be because of I'm on a Mac.

File: Pipfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ python-editor = "==1.0.4"
 pyperclip = "==1.8.0"
 
 [dev-packages]
+pylint = "*"
```
Comment (mikemartino) [inline Pipfile]: Could you just change the dependency it looks like it's pulling down?

```
pylint="==1.9.5"
```

File: README.md [hunk 2]
```diff
@@ -1,7 +1,11 @@
 ## Configure
 ```
+### From source code
 sudo python3 -m pip install pipenv 
-alias standup-notes=<workspace>/standup_notes/bin/standup-notes
+python setup.py install
+
+### From PyPi
+TODO instructions on pip installing from PyPi
```
Comment (mikemartino) [inline README.md]: Are you leaving this for another time?

File: setup.py [hunk 3]
```diff
@@ -12,7 +12,16 @@
     long_description=long_description,
     long_description_content_type="text/markdown",
     url="http://www.mikemartino.ca",
-    packages=setuptools.find_packages(),
+    packages=find_packages(),
+    install_requires=['python-editor==1.0.4', 'pyperclip==1.8.0'],
```
Comment (mikemartino) [inline setup.py:16]: I think this line might have been all I was missing. I had learned about `entry_points` and `package_data`, and added them in similarly, but I gave up before understanding how to get `pip` to know what other dependencies were required. Thanks for this.

File: README.md [hunk 4]
```diff
@@ -1,7 +1,11 @@
 ## Configure
 ```
+### From source code
 sudo python3 -m pip install pipenv 
-alias standup-notes=<workspace>/standup_notes/bin/standup-notes
+python setup.py install
```
Comment (mikemartino) [inline README.md]: This command didn't work for me because I needed:
1. To call `python3`; and
2. To use `sudo`.

```
sudo python3 setup.py install
```
So, went to the internet to find some information on _"Should I call `python` or `python3` if I want to run Python 3.x?_ and I found [PEP 394 -- The "python" Command on Unix-Like Systems](https://www.python.org/dev/peps/pep-0394/). If you have some time, give it a quick read. It's pretty interesting to see how much thought has gone into _exactly_ this scenario. 

Here's what I took away (and I could easily be wrong - it's long lol): 
- If we know we want to run Python 3.x, we should call `python3` directly. 
- However, (I think this PEP describes how) this _isn't_ necessary if you've created a virtual environment where you know the only interpreter available _is_ Python 3.x.
----
TLDR; I think this should be, at a minimum:

```
python3 setup.py install
```
and then caveat it (in a comment) with something like
> \# This might need to be called using sudo on *nix systems.

File: standup_notes/__main__.py [hunk 5]
```diff
@@ -11,7 +11,22 @@
...
+def main():
+    parser = ArgumentParser()
+    parser.add_argument('--list', help='List all stand-up notes.', action='store_true')
+    parser.add_argument('--read-today', help='Read today\'s stand-up notes.', action='store_true')
+    parser.add_argument('--read-tomorrow', help='Read tomorrow\'s stand-up notes.', action='store_true')
+    parser.add_argument('--copy-today', help='Copy today\'s stand-up notes to the clipboard.', action='store_true')
+    parser.add_argument('--copy-tomorrow', help='Copy tomorrow\'s stand-up notes to the clipboard.', action='store_true')
+    parser.add_argument('--edit-today', help='Edit today\'s stand-up notes.', action='store_true')
+    parser.add_argument('--edit-tomorrow', help='Edit tomorrow\'s stand-up notes.', action='store_true')
+    args = parser.parse_args()
```
Comment (mikemartino) [inline standup_notes/__main__.py]: As discussed, I think if you change this to `arguments`, everything will work.

**Review by mikemartino — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (monicaruttle) [reply README.md]: Nope, I wanted to do it after getting feedback in this PR.

File: README.md (hunk 4 above)
Comment (monicaruttle) [reply README.md]: Cool, I'll update that to include sudo. I have my python 3 installation in a location that doesn't require sudo to run, but won't hurt to add it in.

File: README.md [hunk 6]
```diff
@@ -1,7 +1,11 @@
 ## Configure
 ```
+### From source code
 sudo python3 -m pip install pipenv 
-alias standup-notes=<workspace>/standup_notes/bin/standup-notes
+sudo python3 setup.py install
```
Comment (mikemartino) [inline README.md]: Looks like you might be able to just do the install for the user:
```
python3 setup.py install --user
```
Side benefit, it doesn't create directories (_build/_, _dist/_, _*egg/_) in the source directory.

**Review by mikemartino — CHANGES_REQUESTED** [review]
Just one more change to the install line. You can just specify it to get installed for the current user.

**Comment by monicaruttle** [comment]
I updated the README with instructions on installing from PyPi, however I haven't been able to get it to work locally. I wonder if it has to do with my python installation. @mikemartino could you give it a try?

**Comment by mikemartino** [comment]
> I updated the README with instructions on installing from PyPi, however I haven't been able to get it to work locally. I wonder if it has to do with my python installation. @mikemartino could you give it a try?

I'll have a look right now.

**Review by mikemartino — APPROVED** [review]

**Comment by mikemartino** [comment]
> > I updated the README with instructions on installing from PyPi, however I haven't been able to get it to work locally. I wonder if it has to do with my python installation. @mikemartino could you give it a try?
> 
> I'll have a look right now.

Tested this with @monicaruttle and we got it working.
